feat(win32): native Windows support, path adapter & cleanup

dorialn68 opened the pull request:
## What
   • Removes all lingering WSL‑specific paths and assumptions  
   • Adds Windows‑aware command adapter that maps common Unix commands  
     to their PowerShell / CMD equivalents  
   • Introduces automatic path rewriter so `/mnt/c/Users/...` → `C:\Users\...`  
   • Adds `rtl-utils.ts` for runtime platform detection  
   • Deletes stray folders (`codex-cli/Doron`, `codex-cli/Downloads`) from history

   ## Why
   Enables Codex CLI to run natively on Windows without WSL.

   ## How
   1. Platform clause injected into system prompt  
   2. Command adapter translates / rewrites any Unix‑style commands  
   3. Safeguard rewrites incorrect POSIX paths  
   4. Tests & lint updated; all pass on Node 22 under Windows 11

   ## Validation
   ```powershell
   npm ci
   npm run build
   codex shell ["dir","C:\\Users\\<user>\\Downloads"]   # ✅ lists files
   codex "list everything in my Downloads folder"       # ✅ NL → dir /s
   ```

   ## Linked issues
   Closes #<issue‑number‑if‑any>

dorialn68 committed:
feat(windows): native Windows support  Doron Aloni

dorialn68 committed:
feat(windows): native Windows support  Doron Aloni

dorialn68 commented:
I have read the CLA Document and I hereby sign the CLA

dorialn68 commented:
recheck

tibo-openai commented:
We moved to bin/codex.js in https://github.com/openai/codex/pull/318, could you rebase and adapt? Would like to land the rest of the changes in this PR, thanks for the contributions!

tibo-openai committed:
Merge branch 'main' into win32-support

tibo-openai committed:
Update raw-exec.ts

tibo-openai commented:
recheck

dorialn68 committed:
fix(raw-exec): use shell wrapper only on Windows; direct spawn on POSIX to restore unit tests

dorialn68 committed:
feat: Implement core Web UI, Win32 fixes, and backend architecture

Significant progress on multiple fronts:

**Front-end (Web UI):**
- Implement new ChatGPT-like layout (thread + sidebar) using Tailwind CSS.
- Add chat input helpers: Stop/Regenerate buttons, Enter/Shift-Enter behavior.
- Implement LocalStorage persistence for messages and drafts.
- Lay groundwork for multi-agent/task UI elements.
- Refine styling, including distinct bubbles for participant types.
- Make WebSocket URLs dynamic for easier deployment.

**Back-end & Win32 Support:**
- Rework raw-exec for Win32 PowerShell wrapper; fix regressions.
- Plan replacement of node-pty with child_process for future Win32 I/O improvements.
- Add CLI echo back to Chat UI for instant feedback.
- Design multi-CLI worker pool, task dispatcher, and memory system architecture.
- Update dev dependencies (Vite proxy, PostCSS).
- Define clear roadmap for next steps.

dorialn68 committed:
docs: Add Windows and Web UI setup guide

codex-maintainers commented:
Thanks for the contribution! We're primarily focused on the [native/Rust implementation of Codex CLI](https://github.com/openai/codex/discussions/1174) and would love to see if this is still an issue on the latest version. If it is, we're happy to take a look at a PR to the `codex-rs/` implementation.
